torch/_subclasses/fake_tensor.py

```diff
@@ -160,26 +160,26 @@ def _is_tensor_constructor(func: OpOverload):
     )
 
 
-def is_fake(x):
-    if isinstance(x, FakeTensor):
+def is_fake(x, fake_mode=None):
```

Comment:
Yes, this could happen when input is a FakeTensor created outside of compiled region e.g. make_fx(torch.func.functionalize(compiled_f))(x). 

But I guess it's mostly to add more safety checks and enforcing an invariant that all fake tensors in the dynamo should has the same fake_mode as current instructionTranslator.

Reply:
My problem right now is that the `is_fake(fake_mode)` API doesn't really make sense. If you want to uphold an invariant that all fake tensors have the same fake_mode as the current instruction translator, then instead of complicating the `is_fake` check we should add an assertion somewhere (maybe post-fakification?) that all FakeTensors have the same fake mode

Reply:
Yeah, that also sounds good to me! I can change the implementation.